dg/media/app_dashboards/js/training_dashboard.js

```diff
@@ -389,39 +581,29 @@ function plot_questionwise_data(data_json, assessment_ids) {
-            // question_mediators_passed_dict['stacking'] = 'normal'
+            question_mediators_passed_dict['yAxis'] = 0;
 
-            question_mediators_dict['data'] = new Array(data_json.length / 2).fill(0.0);
-            question_mediators_passed_dict['data'] = new Array(data_json.length / 2).fill(0.0);
-            question_percent_dict['data'] = new Array(data_json.length / 2).fill(0.0);
+            question_mediators_dict['data'] = new Array(question_array_length).fill(0.0);
+            question_mediators_passed_dict['data'] = new Array(question_array_length).fill(0.0);
+            question_percent_dict['data'] = new Array(question_array_length).fill(0.0);
 
-            for (i = 0; i < data_json.length / 2; i++) {
+            for (i = 0; i < question_array_length; i++) {
                 if (data_json[i]['question__language__id'] == 2) {
                     x_axis.push(data_json[i]['question__text']);
-                    var total_score = data_json[i]['score__sum'] + data_json[i + 15]['score__sum'];
-                    var total_count = data_json[i]['score__count'] + data_json[i + 15]['score__count'];
+                    var total_score = data_json[i]['score__sum'] + (question_offset == 0 ? 0 : data_json[i + question_offset]['score__sum']);
+                    var total_count = data_json[i]['score__count'] + (question_offset == 0 ? 0 : data_json[i + question_offset]['score__count']);
                     var perc = total_score / total_count * 100;
                     question_percent_dict['data'][i] = parseFloat(perc.toFixed(2));
-                    question_mediators_dict['data'][i] = data_json[i]['participant__count'] + data_json[i + 15]['participant__count'];
+                    question_mediators_dict['data'][i] = data_json[i]['participant__count'] + (question_offset == 0 ? 0 : data_json[i + question_offset]['participant__count']);
                     var eng_pass = data_json[i]['participant__count'] * data_json[i]['score__sum'] / data_json[i]['score__count'];
-                    var hin_pass = data_json[i + 15]['participant__count'] * data_json[i + 15]['score__sum'] / data_json[i + 15]['score__count'];
+                    var hin_pass = (question_offset == 0 ? 0 : data_json[i + question_offset]['participant__count'] * data_json[i + question_offset]['score__sum'] / data_json[i + question_offset]['score__count']);
                     question_mediators_passed_dict['data'][i] = parseInt(eng_pass + hin_pass);
-
                 }
             }
-
-            question_dict.push(question_mediators_dict);
             question_dict.push(question_mediators_passed_dict);
-            question_dict.push(question_percent_dict);
 
         } else {
-            var question_mediators_dict = {};
-            var question_mediators_passed_dict = {};
-            var question_percent_dict = {};
-
+            console.log("length2 = " + data_json.length);
             question_mediators_dict['name'] = 'Total Mediators';
```

Comment:
these attributes are same for both picko seekho and documentation, we can move these statements out of the if else blocks and make them common.